sql/plan/alter_event.go

```diff
@@ -0,0 +1,465 @@
+		}
+		na.At = ts
+	} else {
+		every, ok := e[0].(*expression.Interval)
+		if !ok {
+			return nil, fmt.Errorf("expected `*expression.Interval` but got `%T`", e[0])
+		}
+		na.Every = every
+
+		var ts *OnScheduleTimestamp
+		if len(e) > 1 {
+			ts, ok = e[1].(*OnScheduleTimestamp)
+			if !ok {
+				return nil, fmt.Errorf("expected `*OnScheduleTimestamp` but got `%T`", e[1])
+			}
+			if a.Starts != nil {
+				na.Starts = ts
+			} else if a.Ends != nil {
+				na.Ends = ts
+			}
+		}
+
+		if len(e) == 3 {
+			ts, ok = e[2].(*OnScheduleTimestamp)
+			if !ok {
+				return nil, fmt.Errorf("expected `*OnScheduleTimestamp` but got `%T`", e[2])
+			}
+			na.Ends = ts
+		}
+	}
+
+	return &na, nil
+}
+
+// alterEventIter is the row iterator for *CreateEvent.
+type alterEventIter struct {
+	once          sync.Once
+	originalName  string
+	alterSchedule bool
+	alterStatus   bool
+	eventDetails  sql.EventDetails
+	eventDb       sql.EventDatabase
+}
+
+// Next implements the sql.RowIter interface.
+func (c *alterEventIter) Next(ctx *sql.Context) (sql.Row, error) {
+	run := false
+	c.once.Do(func() {
+		run = true
+	})
+	if !run {
+		return nil, io.EOF
+	}
+
+	var eventEnded = false
+	if c.eventDetails.HasExecuteAt {
+		eventEnded = c.eventDetails.ExecuteAt.Sub(c.eventDetails.LastAltered).Seconds() < 0
+	} else if c.eventDetails.HasEnds {
+		eventEnded = c.eventDetails.Ends.Sub(c.eventDetails.LastAltered).Seconds() < 0
+	}
+
+	if eventEnded {
+		// If the event execution/end time is altered and in the past.
+		if c.alterSchedule {
+			if c.eventDetails.OnCompletionPreserve {
+				// If ON COMPLETION PRESERVE is defined, the event is disabled.
+				c.eventDetails.Status = EventStatus_Disable.String()
+				ctx.Session.Warn(&sql.Warning{
+					Level:   "Note",
+					Code:    1544,
+					Message: fmt.Sprintf("Event execution time is in the past. Event has been disabled"),
+				})
+			} else {
+				return nil, fmt.Errorf("Event execution time is in the past and ON COMPLETION NOT PRESERVE is set. The event was not changed. Specify a time in the future.")
+			}
+		}
+
+		if c.alterStatus {
+			if c.eventDetails.OnCompletionPreserve {
+				// If the event execution/end time is in the past and is ON COMPLETION PRESERVE, status must stay as DISABLE.
+				c.eventDetails.Status = EventStatus_Disable.String()
+			} else {
+				// If event status was set to ENABLE and ON COMPLETION NOT PRESERVE, it gets dropped.
+				err := c.eventDb.DropEvent(ctx, c.originalName)
+				if err != nil {
+					return nil, err
+				}
+				return sql.Row{types.NewOkResult(0)}, nil
+			}
+		}
+	}
+
+	var eventDefinition = sql.EventDefinition{
+		Name:            c.eventDetails.Name,
+		CreateStatement: c.eventDetails.CreateEventStatement(),
+		CreatedAt:       c.eventDetails.Created,
+		LastAltered:     c.eventDetails.LastAltered,
+	}
+
+	// TODO: any EVENT STATUS change should also update the branch-specific event scheduling
```

Comment:
(minor) seems like this is more of a dolt-layer `TODO` and may not be useful in the GMS layer.